alerts/plugins/possible_usernames.py

```diff
@@ -0,0 +1,104 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+import json
+import os
+import typing as types
+
+from lib.config import ES
+from mozdef_util.query_models import PhraseMatch, SearchQuery, TermMatch
+from mozdef_util.elasticsearch_client import ElasticsearchClient
+
+
+CONFIG_FILE = os.path.join(
+    os.path.dirname(__file__),
+    'possible_usernames.json')
+
+
+# TODO: Switch to dataclasses when we move to Python 3.7+
+
+
+class Config(types.NamedTuple):
+    '''Container for the configuration required by the plugin.
+    '''
+
+    search_window_hours: int
+    indices_to_search: types.List[str]
+
+    def load(path: str) -> 'Config':
+        '''Attempt to load a `Config` from a JSON file.
+        '''
+
+        with open(path) as cfg_file:
+            return Config(**json.load(cfg_file))
+
+
+class message:
+    '''Alert plugin that attempts to enrich any alert with a new
+    `details.possible_usernames` field containing a list of names of users who
+    have connected to the host described in the alert within some window of
+    time.
+    '''
+
+    def __init__(self):
+        self.registration = ['promisckernel']
+
+        self._config = Config.load(CONFIG_FILE)
+
+        self._es_client = ElasticsearchClient(ES['servers'])
+
+    def onMessage(self, message):
+        summary_parts = message['summary'].split(' ')
+
+        # We expect the summary to be formatted like:
+        # "Promiscuous mode enabled on HOSTNAME [X]"
```

Comment:
I think we should parse out the hostname via the events within the alert (which will most likely be one hostname inside many of the events), instead of parsing it out of the summary field. This way, we could make the plugin accessible to other alerts in the future, by just simply changing the registration

Reply:
I did consider this approach but my concern was that there's a potential, hypothetically, to retrieve a list of unique hostnames of length greater than one following this approach.  If that were the case, would it necessarily (in the case of all alerts) be safe to assume that either the hostname that appears most frequently is the one to choose?

Reply:
Good question, yeah I think that's a good idea, if we do run into a situation with multiple different hostnames, use the most frequent one as what we search on.

Reply:
Got this working in 4c3d2fa